src/caffe/layers/batch_norm_layer.cpp

```diff
@@ -43,197 +56,207 @@ void BatchNormLayer<Dtype>::Reshape(const vector<Blob<Dtype>*>& bottom,
     CHECK_EQ(bottom[0]->shape(1), channels_);
```

Comment:
When `bottom[0]->num_axes() == 1`, access of `bottom[0]->shape(1)` is out of range.
The condition should be `bottom[0]->num_axes() > 1`.